File: RetroGameGauntlet.Forms/Services/IWikipediaSearchService.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using RetroGameGauntlet.Forms.Models.Api;
using RetroGameGauntlet.Forms.ViewModels;

namespace RetroGameGauntlet.Forms.Services
{
    public interface IWikipediaSearchService
    {
        Task<List<WikipediaApiSearchResponseModel.QueryModel.SearchModel>> GetWikipediaLinks(string query);

        Task<List<WikipediaItemViewModel>> GetItemsForQuery(string query);
    }
}

File: RetroGameGauntlet.PCL/Services/IWikipediaSearchService.fs
namespace RetroGameGauntlet.PCL.Services

open RetroGameGauntlet.PCL.Models.API
open RetroGameGauntlet.PCL.ViewModels
open System.Threading.Tasks
open System.Collections.Generic

/// The Wikipedia search service.
type IWikipediaSearchService = 

    /// Returns the list of Wikipedia queries.
    abstract member GetWikipediaLinksAsync: query:string -> Async<IEnumerable<WikipediaQuerySearchModel>>

    /// Returns the list of Wikipedia pages for query.
    abstract member GetItemsForQueryAsync: query:string -> Async<IEnumerable<WikipediaItemViewModel>>